### PR TITLE
Feature/create ukrlp data

### PR DESCRIPTION
**Pull request by philratcliffe**
### What

Created a new Azure function to create UKRLP lookups in Cosmos DB.

### How to review

Upload the compressed HESA XML data and check it populates an empty UKRLP container.

### CODE REVIEW

File: SharedCode/utils.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""Functions shared by Azure Functions"""
```
Comment (nshumoogum) [inline SharedCode/utils.py]: SharedCode/__inti__.py probably should be removed

File: CreateUkrlpBlobTrigger/ukrlp_client.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                xmlns:ukr="http://ukrlp.co.uk.server.ws.v3">
+                <soapenv:Header/>
+                <soapenv:Body>
+                        <ukr:ProviderQueryRequest>
+                                <SelectionCriteria>
+                                        <UnitedKingdomProviderReferenceNumberList>
+                                                <UnitedKingdomProviderReferenceNumber>{ukprn}</UnitedKingdomProviderReferenceNumber>
+                                        </UnitedKingdomProviderReferenceNumberList>
+                                        <CriteriaCondition>OR</CriteriaCondition>
+                                        <StakeholderId>9</StakeholderId>
```
[the remark this answers is not in the document]
Comment (philratcliffe) [reply CreateUkrlpBlobTrigger/ukrlp_client.py]: Yes, as discussed now set StakeholderId and QueryId from env variable.

File: CreateUkrlpBlobTrigger/ukrlp_client.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        soap_req = f"""
+        <soapenv:Envelope
+                xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+                xmlns:ukr="http://ukrlp.co.uk.server.ws.v3">
+                <soapenv:Header/>
+                <soapenv:Body>
+                        <ukr:ProviderQueryRequest>
+                                <SelectionCriteria>
+                                        <UnitedKingdomProviderReferenceNumberList>
+                                                <UnitedKingdomProviderReferenceNumber>{ukprn}</UnitedKingdomProviderReferenceNumber>
```
[the remark this answers is not in the document]
Comment (philratcliffe) [reply CreateUkrlpBlobTrigger/ukrlp_client.py]: Fixed

File: CreateUkrlpBlobTrigger/sample.dat [hunk 4]
```diff
@@ -0,0 +1 @@
+samples-workitems/workitem.txt
```
[the remark this answers is not in the document]
Comment (philratcliffe) [reply CreateUkrlpBlobTrigger/sample.dat]: Done

File: SharedCode/utils.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    return 'dbs/' + cosmosdb_database_id + '/colls/' + cosmosdb_collection_id
+
+
+def get_cosmos_client():
+    cosmosdb_uri = os.environ['AzureCosmosDbUri']
+    cosmosdb_key = os.environ['AzureCosmosDbKey']
+
+    # TODO inherited this, but need to check we are setting it correctly
+    # Should it have more entropy for example?
+    master_key = 'masterKey'
```
[the remark this answers is not in the document]
Comment (philratcliffe) [reply SharedCode/utils.py]: Thanks. Removed comment. Left in func as this func encapsulates the getting of the cosmos client so not need outside.

**Pull request merged**